### PR TITLE
Add gatsby-source-shopify

### PR DESCRIPTION
**Pull request by angeloashmore**
Submitting PR to add `gatsby-source-shopify` per https://github.com/angeloashmore/gatsby-source-shopify/issues/1

`kylemathews`, `m-allanson`, `dschau`, `pieh` have been added to the NPM package as maintainers.

I tried to mirror the structure of other packages in the monorepo, but please let me know if anything needs to be modified. Thanks!

### CODE REVIEW

File: packages/gatsby-source-shopify/.babelrc [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "presets": [
+    ["../../.babel-preset.js"]
```
Comment (DSchau) [inline packages/gatsby-source-shopify/.babelrc]: New change we recently added, but can we make this

```suggestion
    ["babel-preset-gatsby-package"]
```

File: packages/gatsby-source-shopify/.gitignore [hunk 2]
```diff
@@ -0,0 +1,8 @@
+/__tests__
+/yarn.lock
+/*.js
+!index.js
+/gatsby-node.js
```
Comment (DSchau) [inline packages/gatsby-source-shopify/.gitignore]: Aren't each of these caught by the `/*.js`?

File: packages/gatsby-source-shopify/.prettierrc [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (DSchau) [inline packages/gatsby-source-shopify/.prettierrc]: Let's remove this. We should prefer the root level .prettierrc!

File: packages/gatsby-source-shopify/package.json [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    "gatsby-plugin",
+    "gatsby-source-plugin",
+    "shopify"
+  ],
+  "author": "Angelo Ashmore",
+  "license": "MIT",
+  "peerDependencies": {
+    "gatsby": ">2.0.0-alpha"
+  },
+  "devDependencies": {
```
Comment (DSchau) [inline packages/gatsby-source-shopify/package.json]: If you follow the above approach (adding the babel preset) need to run `yarn add babel-preset-gatsby-package --dev`

**Comment by angeloashmore** [comment]
@LekoArts Yup, finally have some time to finish this. Luckily I have some work projects that may utilize Shopify so I can spend some time working on the source plugin if needed.

File: packages/gatsby-source-shopify/.gitignore (hunk 2 above)
Comment (angeloashmore) [reply packages/gatsby-source-shopify/.gitignore]: You're right, I brought that over from my repo but it's totally unnecessary. Removed!

**Comment by angeloashmore** [comment]
@DSchau Sorry this took so long, but I've submitted the changes per your review. I'm not sure how PR reviews work on GitHub. Anything I need to do on my end?

**Comment by LekoArts** [comment]
@DSchau ^ 👍

**Comment by DSchau** [comment]
```
Successfully published:
 - gatsby-source-shopify@2.0.3
```

This package will now be published from the `gatsbyjs/gatsby` monorepo--hopefully this helps alleviate any confusion in the future, and we can get more 👀 on the package to fix any bugs or add new features.

Thanks to all involved!

**Comment by tleunen** [comment]
Does this package has the fixes this one seem to fix?
https://github.com/nikhilag/gatsby-source-shopify

**Comment by DSchau** [comment]
@tleunen unfortunately, I'm not sure. There seems to be some serious fragmentation and confusion here, so we need to resolve this!

@nikhilag could you weigh in as to what fixes are in your fork of the plugin, and whether they make sense to incorporate upstream into this repo?

**Pull request merged**